extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js

```diff
@@ -69,7 +82,11 @@ export class AmpInstallServiceWorker extends AMP.BaseElement {
 
     if (parseUrl(win.location.href).origin == parseUrl(src).origin) {
       this.loadPromise(this.win).then(() => {
-        install(this.win, src);
+        return install(this.win, src);
+      }).then(reg => {
+        if (!reg || !reg.active || reg.active.state != 'activated') {
+          this.maybeInstallUrlRewrite_();
```

Comment:
So, in my tests (I was testing with 3g emulation) the delta here is somewhat noticeable. I've been running into it regularly. I'll see if I can find a way to remove this risk condition.